File: Lib/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace lib
{
    public class Message
    {
        /// <summary>
        /// BindDevice: data is username
        /// sendcopy: data is copyboard content
        /// </summary>
        public enum Action { Error=0,BindDevice = 2, SendCopy = 1 };
        public Action Act { get; set; }
        public string ClipboardContent { get; set; } //clipboard content
        public string Username { get; set; }
        public string ErrorMsg { get; set; }
        public override string ToString()
        {
            return ToJson();
        }
        public string ToEncryJson()
        {
            return Encrypt.EncryptFunc(this.ToJson());
        }
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this,SettingsIgnoreNullValue);
        }
        public byte[] ToBytesUtf8()
        {
            return Encoding.UTF8.GetBytes(this.ToJson());
        }
        static public Message FromJson(string json)
        {
            return JsonConvert.DeserializeObject<Message>(json,SettingsIgnoreNullValue);
        }
        static public Message FromEncryJson(string encryJson)
        {
            return FromJson(Encrypt.DecryptFunc(encryJson));
        }
        static JsonSerializerSettings SettingsIgnoreNullValue = new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore };
    }
}

File: android/app/src/main/java/com/ChenYuyang/ClipboardSharing/NetworkUtil/Message.java
package com.ChenYuyang.ClipboardSharing.NetworkUtil;
import com.alibaba.fastjson.JSON;


public class Message {
    public Message(){;}
    public Message(Action act,String usrname,String clipboardContent){
        setAct(act);
        this.Username = usrname;
        this.ClipboardContent = clipboardContent;
    }
    public String ToJson(){
        return JSON.toJSONString(this);
    }
    public String ToEncryJson(){
        return EncryptHeper.Encrypt(this.ToJson());
    }
    public static Message FromJson(String json){
        return JSON.parseObject(json,Message.class);
    }
    public static Message FromEncryJson(String encryJson){
        return FromJson(EncryptHeper.Decrypt(encryJson));
    }
    public enum Action{
        Error(0),SendCopy(1),BindDevice(2);
        final int value;
        Action(int v){this.value=v;}
        public int getValue(){return value;}
    }

    public int getAct() {
        return Act;
    }

    public void setAct(int act) {
        this.Act = act;
    }
    public void setAct(Action act){
        setAct(act.getValue());
    }

    public String getClipboardContent() {
        return ClipboardContent;
    }

    public void setClipboardContent(String clipboardContent) {
        ClipboardContent = clipboardContent;
    }

    public String getUsername() {
        return Username;
    }

    public void setUsername(String username) {
        Username = username;
    }

    public String getErrorMsg() {
        return ErrorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        ErrorMsg = errorMsg;
    }

    private String ClipboardContent;
    private String Username;
    private String ErrorMsg;
    private int Act;
}
